packages/@lwc/engine-core/src/framework/base-bridge-element.ts

```diff
@@ -121,9 +122,26 @@ export function HTMLBridgeElementFactory(
     const { attributeChangedCallback: superAttributeChangedCallback } = SuperClass.prototype as any;
     const { observedAttributes: superObservedAttributes = [] } = SuperClass as any;
     const descriptors: PropertyDescriptorMap = create(null);
+
+    // present a hint message so that developers are aware that they have not decorated property with @api
+    if (process.env.NODE_ENV !== 'production') {
+        for (let i = 0, len = privateProperties.length; i < len; i += 1) {
+            const propName = privateProperties[i];
+            attributeToPropMap[htmlPropertyToAttribute(propName)] = propName;
+            descriptors[propName] = {
+                get() {
+                    logWarn(`Property ${propName} is not decorated with @api`);
+                },
+                set: createSetter(propName),
```

Comment:
I'm not sure if I should leave this as undefined or call its non public setter. What do you think @nolanlawson @pmdartus

Reply:
This is the behavior when "proxying/bridging" the non decorated setter

<img width="730" alt="image" src="https://github.com/salesforce/lwc/assets/55927613/24d4703a-8a05-408b-82a4-d12ba69f2282">

Reply:
I think the problem with using `createSetter` here is that it changes the behavior – now, private properties effectively behave like public properties when the setter is called.

Instead, I think the setter should emulate the default behavior (as if you had never added it to the `descriptors`) and also log a warning message. It would be great to have a warning message for both the setter and the getter.

To emulate the default behavior, you can use a closure, e.g.:

```js
let foo = undefined
const descriptor = {
  get() {
    // log warning
    return foo
  },
  set(value) {
    // log warning
    foo = value
  }
}
```